File: nCr.cpp
class Solution{
public:
    
    // Formulas 
    // 1. (a+b) %m = a%m + b%m;
    
    int nCr(int n, int r)
    {
        if(n<r) return 0;
        if((n-r)<r)
            r = n-r;

        int M = 1e9+7;
        vector<int> row(n+1, 1);
        
        for(int i = 2; i<=n; i++)
        {
            //updating 'i'th row
            for(int j=min(i-1, r); j>0; j--)
            {
                row[j] = (row[j]%M + row[j-1]%M)%M;
            }
        }
        return row[r];
        
    }
};

File: nCr.py
import math
class Solution:
    def nCr(self, n, r):
        # code here
        if(r>n):
            return 0
        elif(r==1):
            return 1
        else:
            pr1=1
            pr2=1
            for i in range(1,r+1):
                pr1=pr1*n
                n=n-1
            for i in range(1,r+1):
                pr2=pr2*i
        return (pr1//pr2)%int((math.pow(10,9)+7))
